libraries/AP_Baro/AP_Baro_SPx06.cpp

```diff
@@ -0,0 +1,316 @@
+#define SPA06_REG_CALIB_COEFFS_END             0x24
+
+// PRESSURE_CFG_REG
+#define SPx06_PRES_RATE_32HZ				   (0x05 << 4)
+
+// TEMPERATURE_CFG_REG
+#define SPx06_TEMP_USE_EXT_SENSOR              (1<<7)
+#define SPx06_TEMP_RATE_32HZ				   (0x05 << 4)
+
+// MODE_AND_STATUS_REG
+#define SPx06_MEAS_PRESSURE                    (1<<0)  // measure pressure
+#define SPx06_MEAS_TEMPERATURE                 (1<<1)  // measure temperature
+#define SPx06_MEAS_CON_PRE_TEM				   0x07
+
+#define SPx06_MEAS_CFG_CONTINUOUS              (1<<2)
+#define SPx06_MEAS_CFG_PRESSURE_RDY            (1<<4)
+#define SPx06_MEAS_CFG_TEMPERATURE_RDY         (1<<5)
+#define SPx06_MEAS_CFG_SENSOR_RDY              (1<<6)
+#define SPx06_MEAS_CFG_COEFFS_RDY              (1<<7)
+
+// INT_AND_FIFO_CFG_REG
+#define SPx06_PRESSURE_RESULT_BIT_SHIFT        (1<<2)  // necessary for pressure oversampling > 8
+#define SPx06_TEMPERATURE_RESULT_BIT_SHIFT     (1<<3)  // necessary for temperature oversampling > 8
+
+// Don't set oversampling higher than 8 or the measurement time will be higher than 20ms (timer period)
+#define SPx06_PRESSURE_OVERSAMPLING            8
+#define SPx06_TEMPERATURE_OVERSAMPLING         8
+
+#define SPx06_OVERSAMPLING_TO_REG_VALUE(n)     (ffs(n)-1)
+
+#define SPx06_BACKGROUND_SAMPLE_RATE	32
+
+AP_Baro_SPx06::AP_Baro_SPx06(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> dev)
+    : AP_Baro_Backend(baro)
+    , _dev(std::move(dev))
+{
+}
+
+AP_Baro_Backend *AP_Baro_SPx06::probe(AP_Baro &baro,
+                                       AP_HAL::OwnPtr<AP_HAL::Device> dev)
+{
+    if (!dev) {
+        return nullptr;
+    }
+
+    if (dev->bus_type() == AP_HAL::Device::BUS_TYPE_SPI) {
+        dev->set_read_flag(0x80);
+    }
+
+    AP_Baro_SPx06 *sensor = NEW_NOTHROW AP_Baro_SPx06(baro, std::move(dev));
+    if (!sensor || !sensor->_init()) {
+        delete sensor;
+        return nullptr;
+    }
+    return sensor;
+}
+
+bool AP_Baro_SPx06::_init()
+{
+    if (!_dev) {
+        return false;
+    }
+    WITH_SEMAPHORE(_dev->get_semaphore());
+
+    _dev->set_speed(AP_HAL::Device::SPEED_HIGH);
+
+    uint8_t whoami;
+
+// Sometimes SPx06 has init problems, that's due to failure of reading using SPI for the first time. The SPx06 is a dual
+// protocol sensor(I2C and SPI), sometimes it takes one SPI operation to convert it to SPI mode after it starts up.
+    bool is_SPL06 = false;
+    bool is_SPA06 = false;
+
+    for (uint8_t i=0; i<5; i++) {
+        if (_dev->read_registers(SPx06_REG_CHIP_ID, &whoami, 1)) {
+        	switch(whoami){
+			case SPL06_CHIP_ID:
+				_spx06_type = BARO_SPL06;
+				is_SPL06=true;
+				break;
+			case SPA06_CHIP_ID:
+				_spx06_type = BARO_SPA06;
+				is_SPA06=true;
+				break;
+			default:
+				break;
+			}
+        }
+
+		if(is_SPL06 || is_SPA06)
+			break;
+    }
+    
+    if(!is_SPL06 && !is_SPA06) {
+        return false;
+    }
+
+    // read the calibration data
+    uint8_t SPx06_CALIB_COEFFS_LEN;
+	if(_spx06_type == BARO_SPL06)
```

Comment:
```suggestion
	if(_spx06_type == BARO_SPL06) {
```

... we require braces even on one-line bodies - fix them all, please.

... also an indenting problem